easybuild/framework/easyconfig/tweak.py

```diff
@@ -179,7 +179,9 @@ def __repr__(self):
     for (key, val) in tweaks.items():
 
         if isinstance(val, list):
-            regexp = re.compile(r"^(?P<key>\s*%s)\s*=\s*(?P<val>\[(.|\n)*\])\s*$" % key, re.M)
+            # use non-greedy matching for list value using '*?' to avoid including other parameters in match,
+            # and a lookahead assertion (?=...) so next line is either another parameter definition or a blank line
+            regexp = re.compile(r"^(?P<key>\s*%s)\s*=\s*(?P<val>\[(.|\n)*?\])\s*$(?=\n^(\w+\s*=.*|\s*)$)" % key, re.M)
```

Comment:
The last `|` in the lookahead assertion is there to check that the next line is either a parameter definition or a blank line (or the end of the file, see last commit).

Where is it relevant that `\s` also matches `\n`?